### PR TITLE
avoiding comments in config lines

### PR DESCRIPTION
**Pull request by dmaluenda**

### CODE REVIEW

File: scipion/__main__.py [hunk 1]
```diff
@@ -74,7 +74,8 @@ def config2Dict(configFile, varDict):
         # For each section
         for sectionName, section in config.items():
             for variable, value in section.items():
-                varDict[variable] = expandvars(value)
+                # Expanding user and avoiding comments
+                varDict[variable] = expandvars(value).split('#')[0].strip()
```
Comment (pconesa) [inline scipion/__main__.py:78]: I think the expandvars is using value (that should still have the the #).
Probably is better:

varDict[variable] = expandvars(value.split('#')[0].strip())

??

**Review by pconesa — COMMENTED** [review]
Could you verify expansion happens or try my suggestion?

File: scipion/__main__.py (hunk 1 above)
Comment (pconesa) [reply scipion/__main__.py]: If you, @dmaluenda agree with my latest change I'll merge this.

**Comment by delarosatrevin** [comment]
Regarding variables...I think we agreed on the past not to prepend SCIPION_HOME to vars, I'm I correct?
I'm installing conda inside scipion folder and then my activation  command get incorrectly  replaced SCIPION_HOME prefix.  

I know it is not strictly related to this PR, but  I think that maybe it is a good place to discuss/address this.

**Comment by pconesa** [comment]
Well, by default (without a config files), some variables are SCIPION_HOME dependent. This is the case for SCIPION_SOFTWARE, EM_ROOT, ... otherwise, we will be forced to ask the user for those paths.

When generating the config file, I think for those dependent variables we are writing SCIPION_HOME instead of the actual path

Now, in your case.....are you saying that your CONDA_ACTIVATION_CMD has no SCIPION_HOME but it ends up at run time with it? I do not expect this behaviour.

**Comment by delarosatrevin** [comment]
This is the scenario:
* I have my own config file
* I have installed conda inside scipion folder
* `SCIPION_HOME = /usr/local/scipion`
* `CONDA_ACTIVATION_CMD = . /usr/local/scipion/conda/etc/profile.d/conda.sh`
* The previous command get prefix replaced by empty string, so the activation command ends in: `. conda/etc/profile.d/conda.sh`
* The previous is done here: https://github.com/scipion-em/scipion-pyworkflow/blob/devel/pyworkflow/plugin.py#L506

**Comment by pconesa** [comment]
I get it now, thanks. Looking at it.

**Comment by pconesa** [comment]
Conda issue fixed at: https://github.com/scipion-em/scipion-pyworkflow/pull/135

**Review by pconesa — APPROVED** [review]

**Comment by azazellochg** [comment]
This PR appears to remove #... lines in host.conf when running scipion config --update

**Comment by pconesa** [comment]
oh! no! Easy to revert though, if we don't find a fix for all

@dmaluenda

**Comment by pconesa** [comment]
may be we can keep this behavior only for scipion.conf, but not host.conf.

**Comment by pconesa** [comment]
@azazellochg 
I don't how this PR can cause that. The code is only invoked for SCIPION_CONF and SCIPION_LOCAL_CONF.

Is this something that actually happen to you or you just looked at the code and though there is a change for a bug here?

**Comment by pconesa** [comment]
Probably the --update doesn't make any sense for the host.conf related to https://github.com/scipion-em/scipion-pyworkflow/issues/144

@delarosatrevin ?

**Comment by azazellochg** [comment]
@pconesa it's true I haven't looked at this PR's code. I have #... lines in host.conf, ran scipion config --update and they are gone. I though this PR removed all comments from all config files. Must be somewhere else. I guess the same issue was reported by Wolfgang recently.

**Comment by dmaluenda** [comment]
Indeed I think this PR makes just the opposite. It's ignoring what is after a '#' in the config files, but it's not writing anything.

**Pull request merged**